spec/webdrivers/chrome_finder_spec.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe Webdrivers::ChromeFinder do
+  let(:chrome_finder) { described_class }
+  let(:platform) { Webdrivers::System.platform }
+
+  context 'when the user relies on the gem to figure out the location of Chrome' do
+    it 'determines the location correctly based on the current OS' do
+      expect(chrome_finder.send("#{platform}_location")).not_to be_nil
+    end
+  end
+
+  context 'when the user provides a path to the Chrome binary' do
+    it 'uses Selenium::WebDriver::Chrome.path when it is defined' do
+      Selenium::WebDriver::Chrome.path = chrome_finder.send("#{platform}_location")
+      allow(chrome_finder).to receive(:win_location)
+      allow(chrome_finder).to receive(:mac_location)
+      allow(chrome_finder).to receive(:linux_location)
+      chrome_finder.version
+      expect(chrome_finder).not_to have_received(:win_location)
+      expect(chrome_finder).not_to have_received(:mac_location)
+      expect(chrome_finder).not_to have_received(:linux_location)
+    end
+
+    it "uses ENV['WD_CHROME_PATH'] when it is defined" do
+      loc_method = "#{platform}_location"
+      allow(ENV).to receive(:[]).with('WD_CHROME_PATH').and_return(chrome_finder.send(loc_method))
```

Comment:
maybe we should add a `location` method to the finder classes that does the calling of platform specific versions

Reply:
I added `ChromeFinder,location` and made the platform specific location and version methods `private`. Let me know if you think we should keep them public for any reason.